pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -152,6 +153,17 @@ func (r *reconcileHandler) reconcileShards(ctx context.Context) error {
 	for i := range r.vtk.Status.Partitionings {
 		status := &r.vtk.Status.Partitionings[i]
 		status.ServingWrites = allShardsServingWrites(status.ShardNames, r.vtk.Status.Shards)
+		status.Tablets = totalTablets(status.ShardNames, r.vtk.Status.Shards)
+		status.ReadyTablets = totalReadytablets(status.ShardNames, r.vtk.Status.Shards)
+		status.UpdatedTablets = totalUpdatedTablets(status.ShardNames, r.vtk.Status.Shards)
+
+		if status.ServingWrites == corev1.ConditionTrue {
+			if status.ReadyTablets != status.DesiredTablets {
+				r.setConditionStatus(planetscalev2.VitessKeyspaceReady, corev1.ConditionFalse, "NonReadyServingTablets", fmt.Sprintf("Desired serving tablet count: %d; Ready serving tablet count: %d", status.DesiredTablets, status.ReadyTablets))
```

Comment:
We don't need to update the condition message whenever the particular values change. The status values themselves give that info. Also "serving tablets" seems misleading; what we really mean is that the tablets are in the partitioning that's supposed to be serving, but the tablets themselves are not serving (because they aren't ready). Let's just keep the Reason string simple.

```suggestion
				r.setConditionStatus(planetscalev2.VitessKeyspaceReady, corev1.ConditionFalse, "TabletsNotReady", "Some tablet Pods for the serving partitioning are not Ready.")
```

Reply:
Done.